### PR TITLE
Added MS Teams and mail reporter

### PR DESCRIPTION
**Pull request by estruyf**
Added two new reporters:

- Microsoft Teams reporter
- Mail reporter

### CODE REVIEW

File: docs/src/test-reporters-js.md [hunk 1]
```diff
@@ -429,6 +429,8 @@ npx playwright test --reporter="./myreporter/my-awesome-reporter.ts"
 * [Currents](https://www.npmjs.com/package/@currents/playwright)
 * [GitHub Actions Reporter](https://www.npmjs.com/package/@estruyf/github-actions-reporter)
 * [GitHub Pull Request Comment](https://github.com/marketplace/actions/playwright-report-comment)
+* [Mail Reporter](https://www.npmjs.com/package/playwright-mail-reporter)
+* [Microsoft Teams Reporter](https://www.npmjs.com/package/playwright-msteams-reporter)
```
Comment (mxschmitt) [inline docs/src/test-reporters-js.md:433]: Are you interested in moving the Teams reporter under the [Playwright Community Organisation](https://github.com/playwright-community) while maintaining the ownership?

File: docs/src/test-reporters-js.md [hunk 2]
```diff
@@ -429,6 +429,8 @@ npx playwright test --reporter="./myreporter/my-awesome-reporter.ts"
 * [Currents](https://www.npmjs.com/package/@currents/playwright)
 * [GitHub Actions Reporter](https://www.npmjs.com/package/@estruyf/github-actions-reporter)
 * [GitHub Pull Request Comment](https://github.com/marketplace/actions/playwright-report-comment)
+* [Mail Reporter](https://www.npmjs.com/package/playwright-mail-reporter)
```
Comment (mxschmitt) [inline docs/src/test-reporters-js.md:432]: We are not comfortable with the package that requires a Resend API key.

Comment (estruyf) [reply docs/src/test-reporters-js.md:432]: Fair feedback: I have updated the reporter and released a new version. The reporter now uses SMTP settings.

File: docs/src/test-reporters-js.md (hunk 1 above)
Comment (estruyf) [reply docs/src/test-reporters-js.md:433]: Sound fine to me

Comment (mxschmitt) [reply docs/src/test-reporters-js.md:433]: Status: Transferred the repository over to the Playwright Community, ideally we publish now also under @playwright-community under NPM. This name is not available right now, so I reached out to them for assistance.

**Pull request merged**